bool left_exist_rob_flag
bool right_exist_rob_flag
bool back_exist_rob_flag
